compiler/rustc_resolve/src/lib.rs

```diff
@@ -729,6 +729,7 @@ enum AmbiguityKind {
     GlobVsGlob,
     GlobVsExpanded,
     MoreExpandedVsOuter,
+    ExternalCrate,
```

Comment:
This shouldn't be necessary, all ambiguities that can reach other crates are `GlobVsGlob`.

Reply:
This serves as a tag to indicate it originates from an external source. This distinction is necessary as the behavior differs from locally sourced ambiguity in a crate such as:
1. For ambiguity coming from an external source, the `ambiguity` field in ﻿`NameBindingData` will refer to itself, but local ambiguity will not.
2. The changes may introduce breaking changes. If necessary, these can be easily flagged as warnings.